### PR TITLE
Software Test Report

### PR DESCRIPTION
**Pull request by jugonzal07**
First cut at adding a software test report.

### CODE REVIEW

**Comment by jugonzal07** [comment]
Unsure why black formatting required so many changes-- I would think they would've been caught before. Perhaps I'm using an old version?

File: rct229/ruletest_engine/ruletest_engine.py
Comment (jugonzal07) [inline rct229/ruletest_engine/ruletest_engine.py:435]: We should address this to be consistent. Perhaps these were supposed to be undetermined come to think of it. Addressing this would require updating the ruletest spreadsheets and ruletests. Not a major lift.

File: rct229/ruletest_engine/ruletest_engine.py
Comment (jugonzal07) [inline rct229/ruletest_engine/ruletest_engine.py:417]: This is something I could use help with.

File: rct229/reports/ashrae901_2019_software_test_report.py
Comment (yunjoonjung-PNNL) [inline rct229/reports/ashrae901_2019_software_test_report.py:22]: This is a great way to set the values. Another way could be (and can delete line 21 - just as a reference)
```python
report_json = {
"title":  self.title,
"purpose": self.purpose,
 "ruleset": self.ruleset,
"date_run": self.date_run,
"schema_version": self.schema_version,
"rule_tests": [],}
```
Also, depending on your coding style, I'll leave this up to you!

File: rct229/ruletest_engine/ruletest_engine.py
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_engine.py:366]: Could we have a ```TO-DO``` comment here to add more versions of the ruleset in the future?

**Review by yunjoonjung-PNNL — APPROVED** [review]
Great work! I left a few (optional) comments. I'd appreciate it if you looked into these.

File: rct229/reports/ashrae901_2019_software_test_report.py
Comment (jugonzal07) [reply rct229/reports/ashrae901_2019_software_test_report.py:21]: Great to know! I don't have a strong attachment one way or another. I can update this.

File: rct229/ruletest_engine/ruletest_engine.py
Comment (jugonzal07) [reply rct229/ruletest_engine/ruletest_engine.py:366]: Sure it's not a problem to add that. If another ruletest is is ran, you'll note that it raises an exception to let you know it hasn't been set up yet.

**Comment by yunjoonjung-PNNL** [comment]
Juan, For some reason, the ```section6_mater.json``` file only includes 6-1 TCD. Could you check this out?

File: rct229/ruletest_engine/ruletest_engine.py
Comment (jugonzal07) [reply rct229/ruletest_engine/ruletest_engine.py:417]: Rough categories for each section per Weili on call:

18 - HVAC-BaselineHVAC
19 - HVAC-Airside
23 - HVAC-SystemSpecificRequirements

**Comment by jugonzal07** [comment]
> Juan, For some reason, the `section6_mater.json` file only includes 6-1 TCD. Could you check this out?

Thank you YJ, I addressed this and the other issues! I hope that resolves things. Let me know if you are unsuccessful when running this test report.

**Comment by yunjoonjung-PNNL** [comment]
@jugonzal07 I looked into why the unit tests failed. 
For errors from section 19, I believe this is because the ```section19_master.json``` file includes the RMDS whose rules haven't been merged (e.g., 19-4, etc.) or has some errors (Jackson and I have worked together to get these corrected).
For the errors from section 5 (e.g., 5-3), this is because the expected outcome in the RMD was wrong, as we discussed yesterday.

**Comment by jugonzal07** [comment]
> @jugonzal07 I looked into why the unit tests failed. For errors from section 19, I believe this is because the `section19_master.json` file includes the RMDS whose rules haven't been merged (e.g., 19-4, etc.) or has some errors (Jackson and I have worked together to get these corrected). For the errors from section 5 (e.g., 5-3), this is because the expected outcome in the RMD was wrong, as we discussed yesterday.

Thank you! Yes, I forgot the software pytest reads every JSON in the directory, including this new master JSON. I suppose there's two solutions.

1) Exclude the master JSON from testing. This would be ok once we resolve discrepancies between master JSON and the disaggregated JSONs but it's a bit janky to do so.
2) Remove the master JSONs and instead have the software test report read individual rule JSONs but run slower. I will do a test to see how much slower it is to do so later. It'd help solve the redundancy of having a master JSON and individual rule JSONs. If it's not much slower, than perhaps it simplifies the process.

**Review by weilixu — APPROVED** [review]
Tested with master jsons, works great!

There would be changes to the software testing report. We can leave them to another PR after the report schema is finalized.

**Comment by weilixu** [comment]
Some of rules are failing in this branch, we need to fix them first before merging.

**Pull request merged**
